### DISCUSSION TITLE
Update issue

### DISCUSSION THREAD

**Discussion by NigeUK** [Hofknecht/SystemTrayMenu#498] [Q&A] [answered]
Hello.
There seems to be an issue with update 1.3.3.0:

1. When the app opens a window with a link to the download page appears (click the link).
2. The download page includes a link the Microsoft App store (click the link)
3. The Microsoft App Store has no option to update (store says last updated 9th April ).
4. Click the Open button on the App store page
5. Return to Step 1

I downloaded the Windows Installer from the download page. I open the .exe. An instruction appears telling me to read the FAQ. I cannot find any FAQ.

**Comment by Hofknecht** [comment]
@NigeUK 
thank you for the feedback!
the issue is that windows store takes some hours to some days to get updated.
and in this case i recognized a misstake and changed the package this morning again, so should be fixed maybe tomorrow:

![2023-08-26 21_51_14-Window](https://github.com/Hofknecht/SystemTrayMenu/assets/52528841/0af8371f-bfad-4fd2-bb98-75799255020a)

**Comment by sigma415** [comment]
Hello,
I have the same problem as @NigeUK above: it's still not possible to update via Microsoft App Store.

**Reply by NigeUK** [reply]
Same here @Hofknecht 
The Windows Store app has not updated, yet the app still displays the "update" page with a link.
I totally understand that Microsoft are not exactly .... efficient!

**Reply by Hofknecht** [reply]
@NigeUK @sigma415 
Thank you for the feedback!

We see the info when uploading that it takes a few hours to 3 days until it can have passed the certification process or it is then processed and then another 24 hours until it can be updated everywhere. We didn't pay attention to that and we will pay more attention to it in the future. For us, it means that we first upload to the store and only create a release in Github a few days later.

1.3.3 was processed in the store 24 hours ago, means it should work 100% now, is that correct for you?

I made another bugfix and uploaded 1.3.4 to store yesterday, this has now been processed. According to my assumption I can then press release tomorrow for 1.3.4 in github and it should then work in the Windows Store. It would be cool if I got feedback on whether it would work this time.

**Comment by sigma415** [comment]
> 1.3.3 was processed in the store 24 hours ago, means it should work 100% now, is that correct for you?

Negative  :(
In Microsoft App Store it still shows 1.3.2 is installed and there is no update button, only "Open"

**Comment by Hofknecht** [comment]
Thank you for the info,

i can also imagine trough that i uploaded 1.3.4 that rollout of 1.3.3 is stopped, 

in my case 1.3.3 available:
![2023-08-29 10_30_25-Microsoft Store](https://github.com/Hofknecht/SystemTrayMenu/assets/52528841/903c1ba3-3fa3-400a-aa7e-f5d696cacdd0)
![2023-08-29 10_31_13-Microsoft Store](https://github.com/Hofknecht/SystemTrayMenu/assets/52528841/54312216-9843-4438-a75d-490e072d715b)

let's wait until 1.3.4 available 
then we can see how long it can take and can then use the value from this experience in the future

if that doesn't work we can of course also disable the check for updates for the windows store version, but we'll see if that works in the future if we e.g. wait 5 days between release to store and github then it should work in any case i assume

**Comment by sigma415** [comment]
OK, meanwhile it updated automatically to 1.3.3
Let's see, what  will happen in the next days

**Comment by NigeUK** [comment]
Thanks for the updates.
Mine auto-updated to 1.3.3 and there is no "update" message, so Windows Store considers 1.3.3 as the current version.
I will report back when something changes.

**Comment by NigeUK** [comment]
Still 1.3.3.0 and no updates available.

**Comment by sigma415** [comment]
Maybe it really takes up to three (3) days @ Microsoft

see above:
> We see the info when uploading that it takes a few hours to 3 days until it can have passed the certification process or ...

**Reply by NigeUK** [reply]
Just to confirm, version 3.3.0 is the latest current version?

**Comment by Hofknecht** [comment]
29.8 submission was approved -> 6 days ago
Packaging_1.3.4.0_x86_x64_ReleasePackage.msixbundle

only in windows store

i will press release as soon 1.3.4. is delivered trough store

I can somehow imagine that if the last release is too close, the update will take longer. no clue why, but I think we can only wait until 1.3.4 is distributed via windows store, otherwise windows store users will get a message saying there is no new update

**Comment by sigma415** [comment]
Now after several days the windows app ist still on 1.3.3.
"About" says 1.3.3 and app internal update says "You have the newest version ..."
In Github newest release is also 1.3.3.
But in Windows app configuration it shows 1.3.4 !!!
@Hofknecht: There must be something wrong ......  :(

![2023-09-04_20h45_44](https://github.com/Hofknecht/SystemTrayMenu/assets/34886808/5210e03c-f3b3-4c9d-ab7e-fa6f79721226)

![2023-09-04_20h46_48](https://github.com/Hofknecht/SystemTrayMenu/assets/34886808/1afe1f3c-a975-468e-91ad-5a2b810897dd)

Github:

![2023-09-04_20h47_08](https://github.com/Hofknecht/SystemTrayMenu/assets/34886808/7a32a620-1356-483f-a4fc-377ce6853902)

**Comment by Hofknecht** [comment]
now tried again and just submittted Packaging_1.3.5.0_x86_x64_ReleasePackage.msixbundle, needs some days

**Reply by NigeUK** [reply, marked as answer]
Now updated to 1.3.5.0 and Apps shows the same.
Thanks!

**Reply by sigma415** [reply]
Same here. Anything OK for the moment. :)

**Comment by lbdyck** [comment]
Is there a way in the app to determine the current version?

**Reply by NigeUK** [reply]
Right-Click the app icon in the taskbar and there’s a list of options, including ‘check for updates’ and ‘about ….’

**Reply by lbdyck** [reply]
This is what I get when I try that:
![image](https://github.com/Hofknecht/SystemTrayMenu/assets/42328411/61717254-539f-41bd-b32c-5b0e067b7827)

**Reply by lbdyck** [reply]
Found it in the hidden task bar items 
![image](https://github.com/Hofknecht/SystemTrayMenu/assets/42328411/83998052-4324-411a-b0a0-5d1b425ce46b)
